modules/gapi/include/opencv2/gapi/imgproc.hpp

```diff
@@ -222,6 +222,32 @@ namespace imgproc {
         }
     };
 
+    G_TYPED_KERNEL(GYUV2Gray, <GMat(GMat)>, "org.opencv.colorconvert.imgproc.yuvtogray") {
+        static GMatDesc outMeta(GMatDesc in) {
+            GAPI_Assert(in.depth  == CV_8U);
+            GAPI_Assert(in.planar == false);
+            GAPI_Assert(in.size.width  % 2 == 0);
+            GAPI_Assert(in.size.height % 3 == 0);
+
+            return {CV_8U, 1, Size{in.size.width, in.size.height - (in.size.height / 3)}, false};
+        }
+    };
+
+    G_TYPED_KERNEL(GNV12toGray, <GMat(GMat,GMat)>, "org.opencv.colorconvert.imgproc.nv12togray") {
+        static GMatDesc outMeta(GMatDesc inY, GMatDesc inUV) {
+            GAPI_Assert(inY.depth   == CV_8U);
+            GAPI_Assert(inUV.depth  == CV_8U);
+            GAPI_Assert(inY.chan    == 1);
+            GAPI_Assert(inY.planar  == false);
+            GAPI_Assert(inUV.chan   == 2);
+            GAPI_Assert(inUV.planar == false);
+
+            GAPI_Assert(inY.size.width  == 2 * inUV.size.width);
+            GAPI_Assert(inY.size.height == 2 * inUV.size.height);
+            return inY.withType(CV_8U, 1);
+        }
+    };
```

Comment:
And this one is ok to have here - it is public

Reply:
Done